src/phyltr/commands/annotate.py

```diff
@@ -1,73 +1,55 @@
-
-OPTIONS:
-
-    -e, --extract
-        Build a CSV file of information from a treestream, i.e. reverse the
-        standard behaviour
-
-    -f, --file
-        File to read annotations from
-
-    -k, --key
-        Fieldname which corresponds to tree taxon names, used to link lines
-        of the csv file to tree nodes.
-
-    -m, --multiple
-        If set, when --extract is used, information from each tree in the
-        treestream will be added to the file, with a `tree_number` column
-        used to disambiguate.  When not set, information is extracted only
-        from the first tree in the treestream.
-
-    files
-        A whitespace-separated list of filenames to read treestreams from.
-        Use a filename of "-" to read from stdin.  If no filenames are
-        specified, the treestream will be read from stdin.
-"""
-
 import csv
 import sys
 
 from phyltr.commands.base import PhyltrCommand
 from phyltr.plumbing.sinks import NullSink
-from phyltr.utils.phyltroptparse import OptionParser
+from phyltr.utils.misc import dicts_from_csv
 
-class Annotate(PhyltrCommand):
-
-    parser = OptionParser(__doc__, prog="phyltr annotate")
-    parser.add_option('-e', '--extract', default=False, action="store_true", help="Extract data from annotated tree to file.")
-    parser.add_option('-f', '--file', dest="filename", help="File to read/write annotation data from/to.")
-    parser.add_option('-k', '--key', dest="key", help="Name of column in annotation file to match against taxon names")
-    parser.add_option('-m', '--multiple', default=False, action="store_true")
-
-    def __init__(self, filename, key=None, extract=False, multiple=False):
-        self.filename = filename
-        self.key = key
-        self.extract = extract
-        self.multiple = multiple
 
+class Annotate(PhyltrCommand):
+    """Annotate a the trees in a tree stream with information from a file"""
+    __options__ = [
+        (
+            ('-e', '--extract'),
+            dict(
+                default=False,
+                action="store_true",
+                help="Build a CSV file of information from a treestream, i.e. reverse the"
+                     "standard behaviour")),
+        (
+            ('-f', '--file'),
+            dict(dest="filename", help="File to read/write annotation data from/to.")),
+        (
+            ('-k', '--key'),
+            dict(help="Fieldname which corresponds to tree taxon names, used to link lines of the "
+                      "csv file to tree nodes.")),
+        (
+            ('-m', '--multiple'),
+            dict(
+                default=False, action="store_true",
+                help="If set, when --extract is used, information from each tree in the treestream "
+                     "will be added to the file, with a `tree_number` column used to disambiguate. "
+                     "When not set, information is extracted only from the first tree in the "
+                     "treestream.")),
+    ]
```

Comment:
I put the more detailed documentation from the module docstrings into
- the docstring of the `PhyltrCommand` subclass
- and the `help` argument of the individual arguments.

The help formatter of `argparse` adds info about default values and available choices (e.g. for the `lengths` option of other commands). I think this is still fairly readable and makes sure docs are kept in just one place.